contract_manager/scripts/upgrade_evm_entropy_contracts.ts

```diff
@@ -45,41 +57,56 @@ async function main() {
 
   console.log("Using cache file", cacheFile);
 
-  const payloads: Buffer[] = [];
+  // Try to deploy on every chain, then collect any failures at the end. This logic makes it simpler to
+  // identify deployment problems (e.g., not enough gas) on every chain where they occur.
+  const payloads: (Buffer | undefined)[] = [];
```

Comment:
this type automatically gets converted to `Buffer[]` because of how the indexing works in typescript. Maybe you can use some other falsy value like false.
Another way is to have a allSuccessful flag which we can flip and remove the for loop in the end.